source/distributed/tria.cc

```diff
@@ -3845,14 +4162,31 @@ namespace parallel
     unsigned int
     Triangulation<dim, spacedim>::register_data_attach(
       const std::function<std::vector<char>(const cell_iterator &,
-                                            const CellStatus)> &pack_callback)
+                                            const CellStatus)> &pack_callback,
+      const bool returns_variable_size_data)
     {
-      // add new callback_set
-      cell_attached_data.pack_callbacks.push_back(pack_callback);
+      unsigned int handle = numbers::invalid_unsigned_int;
+
+      // Add new callback function to the corresponding register.
+      // To distinguish between the two types of callback functions, the handles
+      // will be encoded. All callback functions that pack fixed size data will
+      // be given an odd handle, those who pack variable size data will be
+      // assigned to an even one.
```

Comment:
something like this comment should go to the place where you un-encode the handles. you may want to reference the current function `register_data_attach` there